client/rest-high-level/src/main/java/org/elasticsearch/client/ml/dataframe/stats/outlierdetection/Parameters.java

```diff
@@ -45,49 +41,85 @@
     public static ConstructingObjectParser<Parameters, Void> PARSER = new ConstructingObjectParser<>("outlier_detection_parameters",
         true,
         a -> new Parameters(
-            (int) a[0],
-            new TreeSet<>((List<String>) a[1]),
-            (boolean) a[2],
-            (double) a[3],
-            (double) a[4],
-            (boolean) a[5]
+            (Integer) a[0],
+            (String) a[1],
+            (Boolean) a[2],
+            (Double) a[3],
+            (Double) a[4],
+            (Boolean) a[5]
         ));
 
     static {
-        PARSER.declareInt(constructorArg(), N_NEIGHBORS);
-        PARSER.declareStringArray(constructorArg(), METHODS);
-        PARSER.declareBoolean(constructorArg(), COMPUTE_FEATURE_INFLUENCE);
-        PARSER.declareDouble(constructorArg(), FEATURE_INFLUENCE_THRESHOLD);
-        PARSER.declareDouble(constructorArg(), OUTLIER_FRACTION);
-        PARSER.declareBoolean(constructorArg(), STANDARDIZATION_ENABLED);
+        PARSER.declareInt(optionalConstructorArg(), N_NEIGHBORS);
+        PARSER.declareStringArray(optionalConstructorArg(), METHOD);
```

Comment:
This will return a `List<String>` even if there is just one element. If this is to just be a string we should  `declareString`